### ISSUE TITLE
Guide for updating existing Mac OS installation

### ISSUE THREAD

**Issue by kelheor** [daliansky/XiaoMi-Pro-Hackintosh#272]
It would be good to have a mini-guide in wiki or link in the description about how to update an existing installation of Mac OS with saving all existing apps/files. 

For example, I have Mojave and want to update to Catalina (or even install the update for Mojave). Simply installing the update from settings not work and I don't know how to unpack update files to change EFI. I also don't know if I could save all existing files/apps if I'll try to install it from an image file, like in the Installation guide in wiki. I tried to search tonymacx86 forums, but each time when I found a question about the update, it wasn't answered (maybe I missed something, but reading 90+ pages is troublesome).

**Comment by matvada** [comment]
@kelheor i had no problem updating from Mojave to Catalina Beta, i think that the problem is in your EFI folder, the files must be **all updated** to the latest version. I had a problem like your moving from High Sierra to Mojave, i was going crazy but finally i found that the problem was my CLOVER folder.
I solved in this way:

1) Download the zip of the last version of all the files from here https://github.com/daliansky/XiaoMi-Pro-Hackintosh/archive/master.zip
2) Unzip the archive and find the CLOVER FOLDER inside EFI.
3) Mount your EFI through Clover Configurartor app (You can download the program Googling it)
4) Move your CLOVER folder inside mounted EFI to the recycle bin (make a backup first)
5) Drag the CLOVER folder you download from the repository inside EFI.
6) Unmount EFI and reboot

<------OR------>

Or if you want to use the easier way you can copy and paste this command in a Terminal window:
bash -c "$(curl -fsSL https://raw.githubusercontent.com/daliansky/XiaoMi-Pro-Hackintosh/master/install.sh)"

After this select number **(1) Update EFI** and it updates the files automatically.

Doing this i solved the problem and i could update the MacOS through System Preferences like a normal Mac.
I hope i was helpful.

**Comment by kelheor** [comment]
> @kelheor i had no problem updating from Mojave to Catalina Beta, i think that the problem is in your EFI folder, the files must be **all updated** to the latest version. I had a problem like your moving from High Sierra to Mojave, i was going crazy but finally i found that the problem was my CLOVER folder.
> I solved in this way:
> 
> 1. Download the zip of the last version of all the files from here https://github.com/daliansky/XiaoMi-Pro-Hackintosh/archive/master.zip
> 2. Unzip the archive and find the CLOVER FOLDER inside EFI.
> 3. Mount your EFI through Clover Configurartor app (You can download the program Googling it)
> 4. Move your CLOVER folder inside mounted EFI to the recycle bin (make a backup first)
> 5. Drag the CLOVER folder you download from the repository inside EFI.
> 6. Unmount EFI and reboot
> 
> <------OR------>
> 
> Or if you want to use the easier way you can copy and paste this command in a Terminal window:
> bash -c "$(curl -fsSL https://raw.githubusercontent.com/daliansky/XiaoMi-Pro-Hackintosh/master/install.sh)"
> 
> After this select number **(1) Update EFI** and it updates the files automatically.
> 
> Doing this i solved the problem and i could update the MacOS through System Preferences like a normal Mac.
> I hope i was helpful.

I tried both methods and still can't update Mac OS. I'm trying to update from 10.14.1 to 10.14.6. After restart I'm getting installing update progress bar, then screen going black. I waited for 15 minutes, then did force reboot and update wasn't installed.

**Comment by Menchen** [comment]
You need wait more, sometime the update bugs and freeze and don’t reboot automatically when it’s needed, but sometime it’s just don’t update progress bar. I would say waiting 30+Min before a force reboot

**Comment by stevezhengshiqi** [comment]
@kelheor Just reboot and choose 'Install macOS~' entry.

**Comment by kelheor** [comment]
Ok, I finally manage to make it updated. First of all, I disconnected all USB devices and installation went more smoothly (without instant black screen). It was rebooted a couple of times, then after some time of installation, it went to black screen again. I thought that I got same issue again, but when I did force reboot and loaded OS, I saw that update was installed.
Now update center suggests me to install Catalina, hope it also will go smoothly.

Thanks for all your advices. Still thinks that those tips (like update EFI or reboot if you have black screen for N minutes and select install Mac OS) should be somewhere in FAQ at least.
